core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -136,6 +158,32 @@ public static UnderFileSystemFactory find(String path) {
     return eligibleFactories;
   }
 
+  /**
+   * Finds all {@link UnderFileSystemFactory} extensions from the extensions directory.
+   *
+   * @return list of extension factories
+   */
+  private static List<UnderFileSystemFactory> findExtensions() {
```

Comment:
so this method will be called every time and creating all factories on every invocation?